src/lib/interactionManager/interaction.ts

```diff
@@ -161,14 +169,15 @@ export class Interaction {
 
     return this.flow
       .handleInteractionToken(token.interactionToken, token.interactionType)
-      .then(() => {
+      .then(res => {
         this.interactionMessages.push(token)
+        return res
```

Comment:
return handling result. Also, this is probably where tokens should be added to storage (e.g. `storageLib.store.interactionToken(token)`)